assisted-inject-processor-dagger2/src/main/java/com/squareup/inject/assisted/dagger2/processor/internal/hilt.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.inject.assisted.dagger2.processor.internal
+
+import com.squareup.inject.assisted.processor.internal.getAnnotation
+import com.squareup.javapoet.AnnotationSpec
+import com.squareup.javapoet.ClassName
+import javax.lang.model.element.TypeElement
+import javax.lang.model.util.Elements
+
+internal fun generateHiltAnnotation(moduleType: TypeElement, elements: Elements): AnnotationSpec? =
+  // If the @AssistedModule class also has an @InstallIn we just mirror the annotation.
+  moduleType.getAnnotation("dagger.hilt.InstallIn")
+    ?.let { AnnotationSpec.get(it) }
+    // If there is no @InstallIn annotation we add the @DisableInstallInCheck annotation, if
+    // it exists.
+    ?: if (elements.getTypeElement("dagger.hilt.migration.DisableInstallInCheck") != null) {
+      AnnotationSpec.builder(ClassName.get("dagger.hilt.migration", "DisableInstallInCheck"))
+        .build()
+    } else {
+      null
+    }
```

Comment:
This function _**I think**_, is more clearly expressed by not using nullability chaining for control flow.
```kotlin
val installIn = moduleType.getAnnotation("dagger.hilt.InstallIn")
if(installIn != null) 
  return AnnotationSpec.get(installIn)

val disableCheck = ClassName.get("dagger.hilt.migration", "DisableInstallInCheck")
if(elements.getTypeElement(disableCheck.canonicalName()) != null)
  return AnnotationSpec.builder(disableCheck).build()

return null
```
I have personally found maintaining code that (over)uses nullability for control flow to be more difficult than if we just spell it out. and in this case really isn't any more verbose.

Bystander opinion of style. Do with this feedback whatever you deem appropriate.

Reply:
It seems more important that it's more readable for maintainers as I won't have to look at this again (hopefully). Will change.